app/services/hyrax/solr_service.rb

```diff
@@ -23,30 +23,36 @@ def select_path
                                    'Use `Hyrax.config.solr_select_path` instead'
       end
 
-      delegate :add, :commit, :count, :delete, :get, :instance, :post, :query, to: :new
+      delegate :add, :commit, :count, :delete, :get, :instance, :post, :query, :delete_by_query, to: :new
     end
 
     # Wraps rsolr get
     # @return [Hash] the hash straight form rsolr
-    def get(query, use_valkyrie: Hyrax.config.query_index_from_valkyrie, **args)
+    def get(query = nil, use_valkyrie: Hyrax.config.query_index_from_valkyrie, **args)
+      # Make Hyrax.config.solr_select_path the default SOLR path
+      solr_path = args.delete(:path) || Hyrax.config.solr_select_path
+      args = args.merge(q: query) unless query.blank?
+
       if use_valkyrie
-        args = args.merge(q: query)
-        valkyrie_index.connection.get(Hyrax.config.solr_select_path, params: args)
+        valkyrie_index.connection.get(solr_path, params: args)
       else
-        args = args.merge(q: query, qt: 'standard')
-        SolrService.instance.conn.get(Hyrax.config.solr_select_path, params: args)
+        args = args.merge(qt: 'standard') unless query.blank?
+        SolrService.instance.conn.get(solr_path, params: args)
```

Comment:
while we're here, we should probably change the hardcoded `SolrService` calls with `self.class`?

Reply:
@no-reply Yes, I've updated it to use `self.class`.